### ISSUE TITLE
checkpoint load causes discontinuous learning rate log 

### ISSUE THREAD

**Issue by iPwnXX** [wenet-e2e/wenet#1840]
when loading checkpoint to continue training, the lr log ( epoch.yaml and tensorboard log) starts from a small value (e.g. 8e-8), instead of continue from lr recorded in last ckp yaml.

![image](https://user-images.githubusercontent.com/40853420/235077165-1a148b2b-d700-4ef0-b54c-123fadbd6659.png)

the log figure above is an example of load ckp at epoch=80.

in `wenet/bin/train.py` , learning rate is  loaded:  `lr = optimizer.param_groups[0]['lr']`. a judgement checking ckp arg is supposed to be added to handle this case?  Though this minimum restarting lr value only affects first batch of training and recovers to normal by schuduler step at next, the log writer still writes this initial lr value at the end of each epoch iteration, resulting in discontinuous lr logging plot.

**Comment by xingchensong** [comment]
could you help to make a PR to fix this ?

**Comment by iPwnXX** [comment]
> could you help to make a PR to fix this ?

I think there are two approaches: 

- Rewrite  save&load_checkpoint functions, add `optimizer.state_dict()` load and save the same as some other pytorch projects did. Save ckp becomes: `torch.save({'model': model.state_dict(), 'optimizer': optimizer.state_dict()})`. In this way the `lr = optimizer.param_groups[0]['lr']` in train.py would work well without further modification. (I adopted this one in my local workspace). the problem it would bring up is that it don't fit with checkpoints saved previous to this modification.
- Simply call once `scheduler.step()` after `scheduler.set_step(step)` when training using a ckp. In this way the  `lr = optimizer.param_groups[0]['lr']` can get normal lr value without further modification. The problem is that it will raise a UserWarning, telling 'opimizer.step() should be called before scheduler.step(), otherwise pytorch will skip the first value of the learning rate schedule' (doesn't affect much though).
